include/seqan/stream/tokenization.h

```diff
@@ -594,23 +594,27 @@ cropOuter(TContainer &cont, TFunctor const &func)
 
 /*!
  * @fn StringSet#strSplit
- * @brief Append a list of the words in the string, using sep as the delimiter string @link StringSet @endlink.
+ * @brief Split a sequence using a delimiter and append the results to a target string set
  *
  * @signature void strSplit(result, sequence[, sep[, allowEmptyStrings[, maxSplit]]]);
  *
- * @param[out] result           The resulting string set.
+ * @param[out] result           The resulting string set (can be any ContainerOfContainer, also STL)
  * @param[in]  sequence         The sequence to split.
  * @param[in]  sep              The splitter to use (default <tt>' '</tt>).
  * @param[in]  allowEmptyString Whether or not to allow empty strings (<tt>bool</tt>, defaults to <tt>true</tt> iff
  *                              <tt>sep</tt> is given).
  * @param[in]  maxSplit         The maximal number of split operations to do if given.
  */
 
-template <typename TString, typename TSpec, typename TSequence, typename TFunctor, typename TSize>
-inline void
-strSplit(StringSet<TString, TSpec> & result, TSequence const &sequence, TFunctor const &sep, bool allowEmptyStrings, TSize maxSplit)
+template <typename TResult, typename TSequence, typename TFunctor, typename TSize>
+inline SEQAN_FUNC_ENABLE_IF(And<Is<ContainerConcept<TResult> >,
+                                Is<ContainerConcept<typename Value<TResult>::Type > > >, void)
+strSplit(TResult & result, TSequence const &sequence, TFunctor const & sep, bool allowEmptyStrings, TSize maxSplit)
 {
     typedef typename Iterator<TSequence const, Standard>::Type TIter;
+    typedef std::conditional_t<Is<StlContainerConcept<TResult>>::VALUE,
+                               TSequence,
+                               decltype(infix(sequence, 0, 1))&&> TResultValue;
```

Comment:
well I think, it doesn't matter for our Infix types as they are returned as temporary objects anyway.